code/renderers/vue3/src/render.ts

```diff
@@ -22,65 +23,81 @@ export const setup = (fn: (app: any) => void) => {
 
 const map = new Map<
   VueRenderer['canvasElement'],
-  { vueApp: ReturnType<typeof createApp>; reactiveArgs: any }
+  {
+    vueApp: ReturnType<typeof createApp>;
+    reactiveArgs: Args;
+  }
 >();
-
-const elementMap = new Map<VueRenderer['canvasElement'], StoryFnVueReturnType>();
-
+let reactiveState: {
+  globals: Globals;
+};
 export function renderToCanvas(
-  { storyFn, forceRemount, showMain, showException, storyContext }: RenderContext<VueRenderer>,
+  { storyFn, forceRemount, showMain, showException, storyContext, id }: RenderContext<VueRenderer>,
   canvasElement: VueRenderer['canvasElement']
 ) {
-  // fetch the story with the updated context (with reactive args)
-  storyContext.args = reactive(storyContext.args);
-  const element: StoryFnVueReturnType = storyFn();
-  elementMap.set(canvasElement, element);
-
-  const props = (element as any).render?.().props;
-  const reactiveArgs = props ? reactive(props) : storyContext.args;
-
   const existingApp = map.get(canvasElement);
+
+  // if the story is already rendered and we are not forcing a remount, we just update the reactive args
   if (existingApp && !forceRemount) {
-    updateArgs(existingApp.reactiveArgs, reactiveArgs);
+    const element = storyFn();
```

Comment:
Ideally, I think we should only call storyFn during `setup`, not during updating or render?

Reply:
Yeap that is exactly what i'm talking about, but because the decorator system and how  Storybook is handling the render is using React concept which is not reactive you have to run all the stack in every change

Reply:
What, if we would get rid of hooks in all the default decorators in storybook, and explain in the docs that hooks can not be used in user based vue decorators, would that solve the problem?

Reply:
Yeah but these hooks are used by Storybook, no Vue developer will use hooks.
Actually Storybook creating 7 layers of decorators which are basically React Components (  Functional components )
that wrap the story whether is vue,preact, or any other renderer, these decorator are responsible of background, theme 
![image](https://user-images.githubusercontent.com/711292/226339911-3c218732-3041-4d6c-8969-a67f4aea853f.png)
to update the DOM, react needs story renderer ( vue, preact, angular ) to re-render all the tree in order for the hooks to update the DOM